XenAdmin/Wizards/CrossPoolMigrateWizard/CrossPoolMigrateWizard.cs

```diff
@@ -239,8 +244,30 @@ protected override void FinishWizard()
 
                 if (wizardMode == WizardMode.Move && IsIntraPoolMove(pair))
                     new VMMoveAction(vm, pair.Value.Storage, target).RunAsync();
-                else 
-                    new VMCrossPoolMigrateAction(vm, target, SelectedTransferNetwork, pair.Value, wizardMode == WizardMode.Copy).RunAsync();
+                else
+                {
+                    var isCopy = wizardMode == WizardMode.Copy;
+
+                    if (_resumeAfterMigrate)
+                    {
+                        var title = VMCrossPoolMigrateAction.GetTitle(vm, target, isCopy);
+                        var startDescription = isCopy ? Messages.ACTION_VM_COPYING: Messages.ACTION_VM_MIGRATING;
+                        var endDescription = isCopy ? Messages.ACTION_VM_COPIED: Messages.ACTION_VM_MIGRATED;
+
+                        var actions = new List<AsyncAction>()
+                        {
+                            new VMCrossPoolMigrateAction(vm, target, SelectedTransferNetwork, pair.Value, isCopy),

```

Comment:
I suggest to put the VMCrossPoolMigrateAction into a variable outside the `if (_resumeAfterMigrate)` block and use this on both cases (as it is essentially the same action).